tests/batch_test.py

```diff
@@ -0,0 +1,105 @@
+import time
+
+import pytest
+
+import easypost
+import tests.fixture as fixture
+
+
+@pytest.mark.vcr()
+def test_batch_create():
+    batch = easypost.Batch.create(shipments=[fixture.one_call_buy_shipment()])
+
+    assert isinstance(batch, easypost.Batch)
+    assert str.startswith(batch.id, "batch_")
+    assert batch.shipments is not None
+
+
+@pytest.mark.vcr()
+def test_batch_retrieve():
+    batch = easypost.Batch.create(shipments=[fixture.one_call_buy_shipment()])
+
+    retrieved_batch = easypost.Batch.retrieve(batch.id)
+
+    assert isinstance(batch, easypost.Batch)
+    assert batch.id == retrieved_batch.id
+
+
+@pytest.mark.vcr()
+def test_batch_all():
+    batches = easypost.Batch.all(page_size=fixture.page_size())
+
+    batches_array = batches["batches"]
+
+    assert len(batches_array) <= fixture.page_size()
+    assert batches["has_more"] is not None
+    assert all(isinstance(batch, easypost.Batch) for batch in batches_array)
+
+
+@pytest.mark.vcr()
+def test_batch_create_any_buy():
+    batch = easypost.Batch.create(shipments=[fixture.one_call_buy_shipment(), fixture.one_call_buy_shipment()])
+
+    assert isinstance(batch, easypost.Batch)
+    assert str.startswith(batch.id, "batch_")
+    assert batch.num_shipments == 2
+
+
+@pytest.mark.vcr()
+def test_batch_buy():
+    shipment_data = fixture.one_call_buy_shipment()
+
+    batch = easypost.Batch.create(shipments=[shipment_data])
+    # Uncomment the following line if you need to re-record the cassette
+    time.sleep(5)  # Wait enough time for the batch to process buying the shipment
+    batch.buy()
+
+    assert isinstance(batch, easypost.Batch)
+    assert batch.num_shipments == 1
+
+
+@pytest.mark.vcr()
+def test_batch_create_scanform():
+    shipment_data = fixture.one_call_buy_shipment()
+
+    batch = easypost.Batch.create(shipments=[shipment_data])
+    batch.buy()
+
+    # Uncomment the following line if you need to re-record the cassette
+    time.sleep(5)  # Wait enough time for the batch to process buying the shipment
```

Comment:
If we've recorded the cassette already, we'll want to comment this out.

Reply:
oops, nice catch!